chewing-editor: init at 0.1.1 Add package `chewing-editor`

ShamrockLee opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions
-->

###### Motivation for this change
chewing-editor is the user phrase editor
for Chewing input method
(one of the most popular bopomofo Chinese input method)
Chewing IM memorize phrases input by the user automatically
or when the hot keys are pressed.
If applied, the commit will enable Chewing user to correct the phrases in the
database or add new ones, which is a crucial functionality of Chewing IM
that haven't been in NixOS.
The corresponded Packaging Request is #89554.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

iblech commented:
(Just a random passer-by.) Awesome, thank you for making NixOS more accessible for users of traditional Chinese! I confirm that the package builds here. :-)

iblech commented on pkgs/top-level/all-packages.nix:
```diff
@@ -3400,6 +3400,8 @@ in
 
   fcitx-configtool = callPackage ../tools/inputmethods/fcitx/fcitx-configtool.nix { };
 
+  chewing-editor = libsForQt5.callPackage ../applications/misc/chewing-editor { };
```
I believe that we have a general recommendation that the entries in all-packages.nix should roughly be alphabetically sorted. (This recommendation does not seem to be followed through rigorously, perhaps someone more experienced can give an indication why.)

ShamrockLee replied:
Thanks for telling me that.
(I'm a newbie here, and it's the first time that I make a PR in NixOS/nixpkgs.
The second commit actually reflects the suggestions from some senior developers on Discord on `default.nix`, or the compilation couldn't have passed.)
But it seems that there are several rounds of "sorted sections" in `all-packages.nix`. Where should I put my code in?

ShamrockLee replied:
I still haven't figure out the order yet, but I have it inserted somewhere between ch- and cl- to make `all-packages.nix` closer to being alphabetically sorted.
Thank you for your advice.

ryantm replied:
Alphabetical order doesn't really matter and it is hard to check/enforce, so it doesn't happen. Just give it your best effort. Don't put stuff at the bottom though, because that is just asking for a merge conflict.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/203

ryantm reviewed (changes requested):
Please rewrite your git history so it is two separate commits. One that adds the package and then one that adds you as the maintainer.

ShamrockLee committed:
maintainer-list: Add ShamrockLee (myself) to maintainer-list

ShamrockLee committed:
chewing-editor: init at 0.1.1

chewing-editor is the user phrase editor
for Chewing input method (a widely used bopomofo Chinese input method)
Chewing IM memorize phrases input by the automatically
or when the hot keys are pressed.
With this package, users will be able to correct/add phrases in the
database. This is one of the fundamental functionality of Chewing IM
and it will be great to have it in NixOS.
The corresponded Packaging Request is #89554.

ShamrockLee commented:
@ryantm 
I have it split now.

ryantm commented:
@ShamrockLee I redid the history to be just two commits and made it so the commit that adds the chewing-editor derivation can live on its own.
